net/socket/socket.c

```diff
@@ -93,23 +94,50 @@ int psock_socket(int domain, int type, int protocol,
   psock->s_conn   = NULL;
   psock->s_type   = type & SOCK_TYPE_MASK;
 
-  /* Get the socket interface */
+#ifdef CONFIG_NET_USRSOCK
+  /* Get the usrsock interface */
 
-  sockif = net_sockif(domain, psock->s_type, psock->s_proto);
-  if (sockif == NULL)
+  sockif = &g_usrsock_sockif;
+  psock->s_sockif = sockif;
+
+  ret = sockif->si_setup(psock);
+
+  /* When usrsock daemon returns -EPROTONOSUPPORT, it means to use
+   * kernel's network stack, so fallback to kernel socket.
+   */
+
+  if (ret == -ENOSYS || ret == -ENOSUP)
```

Comment:
```
if (ret != -ENOSYS || ret != -ENOSUP)
  {
    return ret;
  }
```
and keep the rest code without change,

Reply:
@xiaoxiang781216 san
Thank you for your suggestion!
I applied it for this PR.
Could you review this PR again?